docs/website/contents/for-developers/HardForkCombinatorRollbackMonotonicityProposal.md

```diff
@@ -0,0 +1,285 @@
+
+However, Frisby thinks such block counting (and the related "double-stability" in the existing ledger rules) is strictly undue complexity that should be removed instead of propagated even deeper.
+
+---
+
+# DRAFT NUMBER 1
+
+## Introduction
+
+This document motivates a simplification of the HFC, that would also alleviate the "double stability" constraint on the ledger's governance rules.
+
+Specifically, it suggests that the HFC should not be vetoing the ledger's decision to transition to the next era while that decision is not yet immutable.
+This veto can only happen in the presence of a Chain Growth violation, but it's confusing none-the-less and requires extra implementation complexity in the HFC.
+
+The suggested simplification will necessarily change some behaviors, but notably the behaviors that principally motivated the HFC's block counting in the first place could be preserved by only answering time-related queries using the immutable ledger state.
+
+## Though Experiment: Predictions within a Simplified Node
+
+Consider a simplified node.
+
+- Only node-to-node protocols: no queries, etc.
+- Pure Praos: no hard fork combinator, no Transitional Praos, etc.
+
+### Question 1
+
+Which aspects of the protocol and ledger's future behavior must this node be able to predict from a given ledger state and why?
+
+- Assume it's well-connected and running in a healthy net.
+- Assume k > 0.
+- Let C be the current best valid chain in the net.
+- Let D be the node's currently selected chain, and assume it's worse than C.
+- Let L be the ledger state resulting from the youngest block that is on both C and D, aka their intersection.
+- Let R count the number of blocks after L on D.
+  Observe that there must be more than R blocks after L on C, since C is better than D.
+  (This isn't necessarily true, due to tiebreakers.
+  However, even with tiebreakers, this remains true in the worst case, which is what ultimately matters.)
+- Common Prefix ensures R <= k.
+- L must contain enough information for the node to correctly judge the validity of the k+1th block after L on C.
+- Otherwise the Header-Body Split would prevent the node from adopting C, even though doing so requires at most rolling back k blocks.
+
+That is a complete answer.
+In particular, there is nothing else this simplified node needs to predict.
+
+If we assume Chain Growth in addition to Common Prefix, then we can further refine the answer.
+
+- Chain Growth ensures that every window of scg (ie "the s parameter of the papers' CG property", aka "one stability window") slots will contain at least k blocks on C.
+- Because of the Header-Body Split, L must contain enough information for the node to correctly judge the validity of any header that is no more than 3k/f after the first block after L on C.
```

Comment:
Why do we require that `L` can judge the validity of at least `k + 1` blocks on any chain that forks from it, but judge the validity of `3k / f` slots only? CG ensures we have at least `k` blocks, but we could have *exactly* `k`. Does that mean that we can only switch to chains that have `> k` blocks in `3k/f` slots?

Reply:
Hmmm, I think this is described below:

>  and hence k+1 blocks in total, as necessary.

Are we counting the intersection block as well? If so I'm still confused about the requirement that `L` can validate `k + 1` blocks *after* the intersection :grimacing:

Reply:
We can tick L up to the next block on their chain. And then we can forecast 3k/f from there.

I think that's actually exactly _one slot short_ 🤔 of ensuring L can validate k+1 headers on a chain that does not violate without having to apply any more blocks to L, unless the sparsity of the k headers after the first violate Chain Growth.